notifications_utils/letter_timings.py

```diff
@@ -30,32 +30,51 @@
 )
 
 
+def is_dvla_working_day(datetime_object):
+    return non_working_days_dvla.is_work_day(
+        datetime_object.date(),
+        division=BankHolidays.ENGLAND_AND_WALES,
+    )
+
+
+def is_royal_mail_working_day(datetime_object):
+    return non_working_days_royal_mail.is_work_day(
+        datetime_object.date(),
+        division=BankHolidays.ENGLAND_AND_WALES,
+    )
+
+
 def set_gmt_hour(day, hour):
     return day.astimezone(pytz.timezone("Europe/London")).replace(hour=hour, minute=0).astimezone(pytz.utc)
 
 
-def get_next_work_day(date, non_working_days):
-    next_day = date + timedelta(days=1)
-    if non_working_days.is_work_day(
-        date=next_day.date(),
-        division=BankHolidays.ENGLAND_AND_WALES,
-    ):
-        return next_day
-    return get_next_work_day(next_day, non_working_days)
+def get_offset_working_day(date, *, is_work_day, offset_days):
+    offset_day = date + timedelta(days=offset_days)
+    if is_work_day(offset_day):
+        return offset_day
+    return get_offset_working_day(offset_day, is_work_day=is_work_day, offset_days=offset_days)
```

Comment:
I know it was recursive before, but it'd be nice to not have to use recursion for this.

I also don't think this works as intended - or at least how I expected? I read this as 'get me +x/-x working days from a date'. But it just ignores all the days between today and the delta and doesn't check whether they're working days or not.

Here's a load of data. It does return the right working day for '2 working days before today' - but only because of a probable bug. When line 55 calls, we immediately skip back another set of `offset_days` rather than just +1/-1 day. This is most pronounced in the final example of 'get me the 11 working days ago'. We go from today (29th) back to 18th, which is a Sunday, and when that's not a working day we go back another 11 days to the 7th, a Wednesday, and return that. I don't think that's what we want.

```python
In [19]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-1)
Out[19]: datetime.datetime(2022, 12, 28, 0, 0)

In [20]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-2)
Out[20]: datetime.datetime(2022, 12, 23, 0, 0)

In [21]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-3)
Out[21]: datetime.datetime(2022, 12, 23, 0, 0)

In [22]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-4)
Out[22]: datetime.datetime(2022, 12, 21, 0, 0)

In [23]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-5)
Out[23]: datetime.datetime(2022, 12, 19, 0, 0)

In [24]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-6)
Out[24]: datetime.datetime(2022, 12, 23, 0, 0)

In [25]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-7)
Out[25]: datetime.datetime(2022, 12, 22, 0, 0)

In [26]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-8)
Out[26]: datetime.datetime(2022, 12, 21, 0, 0)

In [27]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-9)
Out[27]: datetime.datetime(2022, 12, 20, 0, 0)

In [28]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-10)
Out[28]: datetime.datetime(2022, 12, 19, 0, 0)

In [29]: get_offset_working_day(datetime(2022, 12, 29), is_work_day=is_dvla_working_day, offset_days=-11)
Out[29]: datetime.datetime(2022, 12, 7, 0, 0)
```

Reply:
Yeah, it was assuming that `offset_days` would always be `1` or `-1`. I’ve fixed that in [e5f45c9](https://github.com/alphagov/notifications-utils/pull/1016/commits/e5f45c97221ef0821b87d6245554977ef5a47db7)

I’ve also added more helper functions, so your API code should be as simple as 
```python
expected_sent_date = get_dvla_working_day_offset_by(today, days=-2)
```

Reply:
Awesome 🎉